### PR TITLE
[BINS-146] Printing panic on Error level in the logs

### PR DESCRIPTION
**Pull request by viktorbenei**
Previously panic was logged on info level.

This PR changes that to print on Error level.

### CODE REVIEW

**Review by trapacska — DISMISSED** [review]
❤️

**Comment by viktorbenei** [comment]
Question: should we log on another log level, above `error`? 

Seems DataDog supports `critical` level above `error` - https://docs.datadoghq.com/agent/troubleshooting/debug_mode/?tab=agentv6v7

**Review by trapacska — DISMISSED** [review]

File: cmd/worker/main.go [hunk 1]
```diff
@@ -16,17 +18,33 @@ import (
...
-func main() {
-	tracer.Start(
-		tracer.WithService("addons-test-worker"),
-	)
-	defer tracer.Stop()
+// panicLoggerForCurrentGoroutine catches and prints
+// panic on the current goroutine.
+// Why only the current goroutine? Because in go recover()
+// can only catch panics in the same goroutine.
+func panicLoggerForCurrentGoroutine(logger *zap.Logger) {
```
Comment (gaborszakacs) [inline cmd/worker/main.go:25]: The naming and the description feels like a small Go tutorial here, not sure this is the right place to educate devs how panic recovery works, I'd probably just go with `panicLogger` and assume people will know (or have means to read more) how panic recovery works.

File: cmd/worker/main.go [hunk 2]
```diff
@@ -16,17 +18,33 @@ import (
...
+	if r := recover(); r != nil {
+		logger.Error(fmt.Sprintf("Panic: %v", r),
+			zap.String("stacktrace", string(debug.Stack())))
+		if err := logger.Sync(); err != nil {
+			log.Printf("Failed to sync logger: %+v", errors.WithStack(err))
+		}
+		os.Exit(1)
```
Comment (gaborszakacs) [inline cmd/worker/main.go:32]: Are you sure we want to exit in this method? This means any other defers will not run, etc. I'd not expect a logger to exit the program for me as a side-effect.

File: cmd/worker/main.go [hunk 3]
```diff
@@ -45,7 +63,10 @@ func main() {
 	messageReceivingStoppedChan := make(chan bool)
 	messageReceivingErrChan := make(chan error)
 
-	go receiveMessages(ctx, logger, config, messageReceivingStoppedChan, messageReceivingErrChan)
+	go func() {
+		defer panicLoggerForCurrentGoroutine(logger)
+		receiveMessages(ctx, logger, config, messageReceivingStoppedChan, messageReceivingErrChan)
```
Comment (gaborszakacs) [inline cmd/worker/main.go:68]: I think `sub.Receive` will spawn go routines from the input func, so we might not achieve here what we wanted (?).

File: cmd/worker/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (viktorbenei) [reply cmd/worker/main.go:25]: Fixed in #125

File: cmd/worker/main.go (hunk 3 above)
Comment (viktorbenei) [reply cmd/worker/main.go:68]: Yes, but :) Should we remove the `defer panicLoggerForCurrentGoroutine(logger)` line here or leave it here?

(The actual recover is handled here https://github.com/bitrise-io/addons-firebase-testlab/pull/124/files#diff-0dbbf4f572be296ffb1a49e1f874efeb726342c537489a8fd44a0ddc78bd7168R26 , this one as you said will not handle the panic in message handling - still, not sure if we should remove this or keep it here, I don't feel it's "bad" to keep it here as well 🤔 )

**Pull request merged**